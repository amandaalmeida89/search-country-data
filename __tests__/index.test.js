import '@testing-library/jest-dom'
import { render, screen, fireEvent } from '@testing-library/react'
import Page from '../src/pages/index'
import { MockedProvider } from '@apollo/client/testing';
import GetCurrencyQuery from '../src/graphql/queries/getCurrency';

window.scrollTo = jest.fn();

const mocks = [
  {
    request: {
      query: GetCurrencyQuery,
      variables: {
        countryName: 'Sweden'
      }
    },
    result: {
      data: {
        countries: [{
          name: 'Sweden',
          currency: 'Swedish krona'
        }
      ]}
    }
  }
];

describe('Page', () => {
  it('renders initial page', async () => {
    const { container } = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Page />
      </MockedProvider>
    );
    expect(await screen.findByText('Currency Search')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should return input value and currency selected', async () => {
    const { container } = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Page />
      </MockedProvider>
    );

    const input = await screen.getByRole('textbox');

    //focus event without value
    await fireEvent.focus(input, { target: {}})

    //blur event without value
    await fireEvent.blur(input, { target: {}})

    //input event value
    await fireEvent.input(input, { target: { value: 'Sweden' }})

    //focus event with value
    await fireEvent.focus(input, { target: {}})

    //click event in currency collapse list
    const list = await screen.findByText('Swedish krona')
    await fireEvent.click(list)

    expect(input).toHaveValue('Sweden')
    expect(container).toMatchSnapshot();
  });
});