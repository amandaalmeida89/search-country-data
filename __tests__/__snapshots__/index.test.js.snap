// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders initial page 1`] = `
<div>
  <div
    class="css-hboir5"
  >
    <div
      class="chakra-container css-s2ohfa"
    >
      <div
        class="css-1rkksne"
      >
        <h2
          class="chakra-heading css-5bia5c"
        >
          Search country data
        </h2>
      </div>
      <div
        class="css-gmuwbf"
      >
        <div
          class="css-1wt0574"
        >
          <b
            class="chakra-text css-1108u5m"
          >
            From
          </b>
          <div
            class="css-2w322j"
          >
            <div
              class="chakra-input__group css-115pn5h"
              data-group="true"
            >
              <input
                class="chakra-input css-0"
                id="ds-input"
                placeholder="Country Name"
                value=""
              />
              <div
                class="chakra-input__right-element css-dp9r5f"
              >
                <svg
                  class="chakra-icon css-14nh4z6"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
            <div
              class="chakra-collapse"
              style="overflow: hidden; display: none; opacity: 0; height: 0px;"
            >
              <ul
                class="css-h2imyi"
                role="list"
              >
                <p
                  class="chakra-text css-ax9bdp"
                />
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-sq91g9"
    >
      <div
        class="chakra-wrap css-1vae7t2"
      >
        <ul
          class="chakra-wrap__list css-19lo6pj"
        >
          <li
            class="chakra-wrap__listitem css-1yp4ln"
          >
            <div
              class="css-gmuwbf"
            >
              <p
                class="chakra-text css-1it6ou0"
              >
                Copyright Â© 
                2024
                 - by Amanda Nascimento
              </p>
              <a
                href="https://github.com/amandaalmeida89/datastory-code-challenge"
                target="new"
              >
                <img
                  alt="github Logo"
                  data-nimg="1"
                  decoding="async"
                  height="30"
                  loading="lazy"
                  src="/_next/image?url=%2Fgithub.png&w=64&q=75"
                  srcset="/_next/image?url=%2Fgithub.png&w=32&q=75 1x, /_next/image?url=%2Fgithub.png&w=64&q=75 2x"
                  style="color: transparent;"
                  width="30"
                />
              </a>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`Page should return input value and currency selected 1`] = `
<div>
  <div
    class="css-hboir5"
  >
    <div
      class="chakra-container css-s2ohfa"
    >
      <div
        class="css-1rkksne"
      >
        <h2
          class="chakra-heading css-5bia5c"
        >
          Search country data
        </h2>
      </div>
      <div
        class="css-gmuwbf"
      >
        <div
          class="css-1wt0574"
        >
          <b
            class="chakra-text css-1108u5m"
          >
            From
          </b>
          <div
            class="css-2w322j"
          >
            <div
              class="chakra-input__group css-115pn5h"
              data-group="true"
            >
              <input
                aria-readonly="true"
                class="chakra-input css-0"
                id="ds-input"
                placeholder="Country Name"
                readonly=""
                value="Sweden"
              />
              <span
                class="css-1madzyw"
              >
                <span
                  class="css-zvlevn"
                >
                  Sweden
                </span>
                <button
                  aria-label="close"
                  class="css-1clv9wk"
                  type="button"
                >
                  <svg
                    class="chakra-icon css-1md6f3y"
                    focusable="false"
                    viewBox="0 0 512 512"
                  >
                    <path
                      d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
              </span>
              <div
                class="chakra-input__right-element css-dp9r5f"
              >
                <svg
                  class="chakra-icon css-14nh4z6"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
            <div
              class="chakra-collapse"
              style="overflow: hidden; display: block; opacity: 1; height: 0px;"
            >
              <ul
                class="css-h2imyi"
                role="list"
              >
                <li
                  class="css-1kztbxf"
                >
                  <div
                    class="css-k008qs"
                  >
                    <b
                      class="chakra-text css-1cq884l"
                    >
                      Sweden
                    </b>
                    Â 
                    <p
                      class="chakra-text css-1cq884l"
                    />
                  </div>
                </li>
              </ul>
            </div>
          </div>
          <div
            class="css-1wmnidt"
          >
            <div
              class="css-oc1vt3"
            >
              <b
                class="chakra-text css-1cq884l"
              >
                Currency:
              </b>
              Â 
              <p
                class="chakra-text css-1cq884l"
              >
                Swedish krona
              </p>
            </div>
            <div
              class="css-oc1vt3"
            >
              <b
                class="chakra-text css-1cq884l"
              >
                PhoneCode:
              </b>
              Â 
              <p
                class="chakra-text css-1cq884l"
              >
                46
              </p>
            </div>
            <div
              class="css-oc1vt3"
            >
              <b
                class="chakra-text css-1cq884l"
              >
                Symbol:
              </b>
              Â 
              <p
                class="chakra-text css-1cq884l"
              >
                Skr
              </p>
            </div>
            <div
              class="css-oc1vt3"
            >
              <b
                class="chakra-text css-1cq884l"
              >
                Flag:
              </b>
              Â 
              <p
                class="chakra-text css-1cq884l"
              >
                ðŸ‡¸ðŸ‡ª
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-sq91g9"
    >
      <div
        class="chakra-wrap css-1vae7t2"
      >
        <ul
          class="chakra-wrap__list css-19lo6pj"
        >
          <li
            class="chakra-wrap__listitem css-1yp4ln"
          >
            <div
              class="css-gmuwbf"
            >
              <p
                class="chakra-text css-1it6ou0"
              >
                Copyright Â© 
                2024
                 - by Amanda Nascimento
              </p>
              <a
                href="https://github.com/amandaalmeida89/datastory-code-challenge"
                target="new"
              >
                <img
                  alt="github Logo"
                  data-nimg="1"
                  decoding="async"
                  height="30"
                  loading="lazy"
                  src="/_next/image?url=%2Fgithub.png&w=64&q=75"
                  srcset="/_next/image?url=%2Fgithub.png&w=32&q=75 1x, /_next/image?url=%2Fgithub.png&w=64&q=75 2x"
                  style="color: transparent;"
                  width="30"
                />
              </a>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
